Homework2.Tests/ArrayExtensionsTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Linq;
+using Xunit;
+
+namespace Homework2.Tests
+{
+    public class ArrayExtensionsTests
+    {
+        [Theory]
+        [InlineData(999, 4)]
+        [InlineData(999, 0)]
+        [InlineData(999, 10)]
+        public void AddAt_adds_at_idx(int item, int idx)
```

Comment:
The assignment didn't ask for bounds checking (or even testing) so you're already above and beyond.  If you were going for real world application you could add a test for adding an item to an empty array or a null array, or an empty index or an index beyond the length of the data.

Reply:
Those tests definitely should be there. Since I did my submission late at night I tried to stick with the happy path tests though... Maybe in future submissions my tests will be much better :D